tests/client_test.go

```diff
@@ -955,3 +968,53 @@ func compareStreamUpstreamServers(x []client.StreamUpstreamServer, y []client.St
 
 	return reflect.DeepEqual(xServers, yServers)
 }
+
+func TestUpstreamServerWithDrain(t *testing.T) {
+	httpClient := &http.Client{}
+	c, err := client.NewNginxClient(httpClient, "http://127.0.0.1:8080/api")
+	if err != nil {
+		t.Fatalf("Error connecting to nginx: %v", err)
+	}
+
+	server := client.UpstreamServer{
+		Server:      "127.0.0.1:2000",
+		MaxFails:    &defaultMaxFails,
+		FailTimeout: "10s",
+		SlowStart:   "0s",
+		Weight:      &defaultWeight,
+	}
+
+	// Add upstream server
+	err = c.AddHTTPServer(upstream, server)
+	if err != nil {
+		t.Errorf("Error adding upstream server: %v", err)
+	}
+
+	// Get existing upstream servers
+	servers, err := c.GetHTTPServers(upstream)
+	if err != nil {
+		t.Fatalf("Error getting HTTPServers: %v", err)
+	}
+
+	if len(servers) != 1 {
+		t.Errorf("Too many servers")
+	}
+
+	// Setting drain directive to true
+	servers[0].Drain = true
+
+	// Updating existing upstream server with drain directive
+	_, _, err = c.UpdateHTTPServers(upstream, servers)
```

Comment:
`UpdateHTTPServers` doesn't update the existing upstream server - it can only add or delete, but not update. As a result, this test is not meaningful.

If we want to have a proper test, the client needs to have a method that updates an existing server. This method will send a PATCH request to the API http://nginx.org/en/docs/http/ngx_http_api_module.html#http_upstreams_http_upstream_name_servers_http_upstream_server_id to update an upstream server. However, this should not be in the scope of this tasks. For one thing, that new method will require new tests as well.

To fix this test, I suggest the following:
1. Add a test upstream to `test.conf`:
```
upstream test-drain {
    zone test-drain 64k;

    server 127.0.0.1:9001 drain;
}
```
2. Change the test so that:
  a. the test gets the servers of that upstreams
  b. it checks that the number of upstream servers is 1
  c. it compares the upstream server with the expected. The expected should have drain = `true`.

This way the updated test checks that if the drain is set to true in the API, the client will see the same value in the API response. However, it doesn't check that the client can set drain `true`, because of the mentioned limitation above.